### PR TITLE
NOTICK Transaction verification tests

### PR DESCRIPTION
**Pull request by lankydan**

### CODE REVIEW

**Review by vlajos — DISMISSED** [review]
LGTM.

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoTransactionEncumbranceVerifier.kt [hunk 1]
```diff
@@ -2,67 +2,73 @@ package net.corda.ledger.utxo.flow.impl.transaction.verifier
 
 import net.corda.v5.crypto.SecureHash
 import net.corda.v5.ledger.utxo.ContractVerificationFailure
-import net.corda.v5.ledger.utxo.StateAndRef
+import net.corda.v5.ledger.utxo.transaction.UtxoLedgerTransaction
 
+class UtxoTransactionEncumbranceVerifier(private val transaction: UtxoLedgerTransaction) {
```
Comment (blsemo) [inline components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoTransactionEncumbranceVerifier.kt]: What's the point of creating a class when it does not encapsulate any data beyond the one function call?

**Review by MrMatthewLayton — DISMISSED** [review]
LGTM

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoTransactionEncumbranceVerifier.kt (hunk 1 above)
Comment (MrMatthewLayton) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoTransactionEncumbranceVerifier.kt]: @blsemo The Single Responsibility Principle; the caller likely shouldn't be responsible for this verification.

Comment (blsemo) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoTransactionEncumbranceVerifier.kt]: Ok but it's still a single function you call, you just wrap it in a class? 
Maybe just a question of programming language/cultural background - where I'm coming from, the rule was that anything that doens't need access to the internals of a class (or is part of its interface) should not be a member function, but a free function.
We still got single responsibility - it's that functions responsibilty to do the check, just not a class wrapping it.

Comment (MrMatthewLayton) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoTransactionEncumbranceVerifier.kt]: @blsemo Yes that makes sense, we could just have top-level functions for this.

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoTransactionBuilderVerifier.kt [hunk 2]
```diff
@@ -45,7 +44,7 @@ class UtxoTransactionBuilderVerifier(private val transactionBuilder: UtxoTransac
     }
 
     private fun verifyEncumbranceGroups() {
-        check(transactionBuilder.getEncumbranceGroups().all { it.value.size > 1 }) {
+        check(transactionBuilder.getEncumbranceGroups().all { it.value.isNotEmpty() }) {
```
Comment (MrMatthewLayton) [inline components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoTransactionBuilderVerifier.kt]: Encumbrance groups must contain > 1 item.

**Review by MrMatthewLayton — CHANGES_REQUESTED** [review]

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoTransactionBuilderVerifier.kt (hunk 2 above)
Comment (lankydan) [reply components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/verifier/UtxoTransactionBuilderVerifier.kt]: Ah ok, I misunderstood that code then.

**Review by MrMatthewLayton — APPROVED** [review]

**Pull request merged**